src/neo4j_genai/experimental/pipeline/pipeline.py

```diff
@@ -306,22 +242,79 @@ async def next(
             yield next_node
         return
 
+    async def get_input_config_for_task(self, task: TaskPipelineNode) -> dict[str, str]:
+        """Build input definition for a given task.,
+        The method needs to access the input defs defined in the edges
+        between this task and its parents.
+
+        Args:
+            task (TaskPipelineNode): the task to get the input config for
+
+        Returns:
+            dict: a dict of
+                {input_parameter: path_to_value_in_the_pipeline_results_dict}
+        """
+        input_config: dict[str, Any] = {}
+        # make sure dependencies are satisfied
+        # and save the inputs defs that needs to be propagated from parent components
+        for prev_edge in self.pipeline.previous_edges(task.name):
+            prev_node = self.pipeline.get_node_by_name(prev_edge.start)
+            prev_status = await prev_node.read_status()
+            if prev_status != RunStatus.DONE:
+                logger.critical(f"Missing dependency {prev_edge.start}")
+                raise PipelineMissingDependencyError(f"{prev_edge.start} not ready")
+            if prev_edge.data:
+                prev_edge_data = prev_edge.data.get("input_config") or {}
```

Comment:
Super minor but you can specify a default on `.get()` 
```suggestion
                prev_edge_data = prev_edge.data.get("input_config", {})
```